### PR TITLE
Create pair from newly created governor token

### PR DESCRIPTION
**Pull request by akhilesh-singh-Lab49**
Generate governor token from proposal and create pair with it.

### CODE REVIEW

File: contracts/common/GovernorTokenCreate.sol [hunk 1]
```diff
@@ -101,13 +102,19 @@ contract GovernorTokenCreate is GovernorCountingSimpleInternal {
 
         (responseCode, tokenAddress) = createFungibleToken(
             newToken,
-            uint256(0),
+            uint256(10000000000000000),
```
Comment (amandeeprehal-lab49) [inline contracts/common/GovernorTokenCreate.sol:105]: Why this change?

File: integrationTest/governor.ts [hunk 2]
```diff
@@ -130,6 +138,166 @@ const quorumReached = async (proposalId: BigNumber) => {
...
+    .execute(client);
+
+  const receipt = await contractTokenTx.getReceipt(client);
+  const record = await contractTokenTx.getRecord(client);
+  const tokenAddress = record.contractFunctionResult!.getAddress(0);
+  
+  console.log(
+    `quorumReached tx status ${receipt.status}}`
+  );
+  return {tokenAddress};
```
Comment (amandeeprehal-lab49) [inline integrationTest/governor.ts:161]: you can simply return address why as object.

`return {tokenAddress};`

to

`return tokenAddress;`

File: integrationTest/governor.ts [hunk 3]
```diff
@@ -130,6 +138,166 @@ const quorumReached = async (proposalId: BigNumber) => {
...
+                                        .addAddress(baseContract.address)
+  const contractSetPairsTx = await new ContractExecuteTransaction()
+    .setContractId(factoryContractId)
+    .setFunction("setUpFactory", contractFunctionParameters)
+    .setGas(9000000)
+    .execute(client);
+  const contractSetPairRx = await contractSetPairsTx.getReceipt(client);
+  const response = await contractSetPairsTx.getRecord(client);
+  const status = contractSetPairRx.status;
+  console.log(`\nSetupFactory Result ${status} code: ${response.contractFunctionResult!.getAddress()}`);
```
Comment (amandeeprehal-lab49) [inline integrationTest/governor.ts:177]: Pls format the code.

File: integrationTest/governor.ts [hunk 4]
```diff
@@ -130,6 +138,166 @@ const quorumReached = async (proposalId: BigNumber) => {
...
+};
+
+const createPair = async (contractId: string | ContractId, token0: string, token1: string) => {
+  
+  console.log(
+    `createPair TokenA TokenB`
+  );
+  const addLiquidityTx = await new ContractExecuteTransaction()
```
Comment (amandeeprehal-lab49) [inline integrationTest/governor.ts:185]: Use correct name,I think you are creating pair!

File: integrationTest/governor.ts [hunk 5]
```diff
@@ -130,6 +138,166 @@ const quorumReached = async (proposalId: BigNumber) => {
...
+    .sign(treasureKey);
+
+  const addLiquidityTxRes = await addLiquidityTx.execute(client);
+  const transferTokenRx = await addLiquidityTxRes.getReceipt(client);
+  const transferTokenRecord = await addLiquidityTxRes.getRecord(client);
+  const contractAddress = transferTokenRecord.contractFunctionResult!.getAddress(0);
+  console.log(`CreatePair address: ${contractAddress}`);
+  console.log(`CreatePair status: ${transferTokenRx.status}`);
+  return contractAddress;
+  //return `0x${contractAddress}`;
```
Comment (amandeeprehal-lab49) [inline integrationTest/governor.ts]: why this?

File: integrationTest/governor.ts [hunk 6]
```diff
@@ -130,6 +138,166 @@ const quorumReached = async (proposalId: BigNumber) => {
...
+};
+
+const getAllPairs = async (): Promise<string> => {
+  console.log(
+    `getAllPairs`
+  );
+  const liquidityPool = await new ContractExecuteTransaction()
```
Comment (amandeeprehal-lab49) [inline integrationTest/governor.ts:212]: pls correct names

File: integrationTest/governor.ts [hunk 7]
```diff
@@ -130,6 +138,166 @@ const quorumReached = async (proposalId: BigNumber) => {
...
+  const liquidityPoolTx = await liquidityPool.execute(client);
+  const response = await liquidityPoolTx.getRecord(client);
+   console.log(`getPairs: ${response.contractFunctionResult!.getAddress(0)}`);
+  const transferTokenRx = await liquidityPoolTx.getReceipt(client);
+  console.log(`getPairs: ${transferTokenRx.status}`);
+  return response.contractFunctionResult!.getAddress(0);
+};
+
+const upgradeTo = async (newImplementation: string) => {
+  const liquidityPool = await new ContractExecuteTransaction()
```
Comment (amandeeprehal-lab49) [inline integrationTest/governor.ts:229]: Same here pls

File: contracts/common/GovernorTokenCreate.sol (hunk 1 above)
Comment (akhilesh-singh-Lab49) [reply contracts/common/GovernorTokenCreate.sol:105]: I was test by setting Initial supply for new token, removed.

**Review by amandeeprehal-lab49 — APPROVED** [review]
